cmd/wallet-sdk-gomobile/docs/usage.md

```diff
@@ -0,0 +1,401 @@
+##### Kotlin (Android)
+
+```kotlin
+import dev.trustbloc.wallet.sdk.api.*
+import dev.trustbloc.wallet.sdk.did.*
+import dev.trustbloc.wallet.sdk.localkms
+
+val kms = localkms.KMS(null) // The null store argument causes it to use in-memory storage
+
+val keyHandle = kms.create(localkms.KeyTypeED25519)
+
+val didCreator = did.NewCreatorWithKeyReader(kms)
+
+val createDIDOpts = api.CreateDIDOpts()
+createDIDOpts.setKeyID = keyHandle.getKeyID()
+createDIDOpts.setVerificationType = did.Ed25519VerificationKey2018
+
+val didDocResolution = didCreator.create("key", createDIDOpts) // Create a did:key doc
+```
+
+##### Swift (iOS)
+
+```swift
+import Walletsdk
+
+let kms = LocalkmsNewKMS(nil) // The nil store argument causes it to use in-memory storage
+
+let keyHandle = kms.create(LocalkmsKeyTypeED25519)
+
+let didCreator = DidNewCreatorWithKeyReader(kms)
+
+let createDIDOpts = ApiCreateDIDOpts(keyID: keyHandle.keyID, verificationType: DidEd25519VerificationKey2018)
+
+let didDocResolution = didCreator.create("key", createDIDOpts) // Create a did:key doc
+```
+
+## DID Resolver
+
+### Examples
+
+#### Kotlin (Android)
+
+```kotlin
+import dev.trustbloc.wallet.sdk.did.*
+
+val didResolver = did.Resolver()
+
+val didDoc = didResolver.resolve("did:key:z6MkhaXgBZDvotDkL5257faiztiGiC2QtKLGpbnnEGta2doK")
+```
+
+#### Swift (iOS)
+
+```swift
+import Walletsdk
+
+let didResolver = DidNewResolver()
+
+let didDoc = didResolver.resolve("did:key:z6MkhaXgBZDvotDkL5257faiztiGiC2QtKLGpbnnEGta2doK")
+```
+
+### With Key Reader
+The keys used for DID documents are not automatically generated on the caller's behalf.
+They must be passed in by the caller.
+
+#### Examples
+
+##### Kotlin (Android)
+
+```kotlin
+import dev.trustbloc.wallet.sdk.api.*
+import dev.trustbloc.wallet.sdk.did.*
+import dev.trustbloc.wallet.sdk.localkms
+
+val kms = localkms.KMS(null) // The null store argument causes it to use in-memory storage
+
+val keyHandle = kms.create(localkms.KeyTypeED25519)
+
+val didCreator = did.NewCreatorWithKeyReader(kms)
+
+val createDIDOpts = api.CreateDIDOpts()
+createDIDOpts.setKeyID = keyHandle.getKeyID()
+createDIDOpts.setVerificationType = did.Ed25519VerificationKey2018
+
+val didDocResolution = didCreator.create("key", createDIDOpts) // Create a did:key doc
+```
+
+##### Swift (iOS)
+
+```swift
+import Walletsdk
+
+let kms = LocalkmsNewKMS(nil) // The nil store argument causes it to use in-memory storage
+
+let keyHandle = kms.create(LocalkmsKeyTypeED25519)
+
+let didCreator = DidNewCreatorWithKeyReader(kms)
+
+let createDIDOpts = ApiCreateDIDOpts(keyID: keyHandle.keyID, verificationType: DidEd25519VerificationKey2018)
+
+let didDocResolution = didCreator.create("key", createDIDOpts) // Create a did:key doc
+```
+
+## OpenID4CI
+
+The OpenID4CI package contains an API that can be used by a [holder](https://www.w3.org/TR/vc-data-model/#dfn-holders)
+to go through the [OpenID4CI](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html) flow.
```

Comment:
change links to spec according to this https://github.com/trustbloc/vcs/tree/main/component/wallet-cli#oidc4cioidc4vc-flow

Links you used keeps changing every month, we can not always keep up with those updates.

Better we mention exact version we choose to implement.

Reply:
Done!